<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Curation of Bioconductor package metadata, targeting EDAM ontology and ELIXIR bio.tools metadata schemas • biocEDAM</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Curation of Bioconductor package metadata, targeting EDAM ontology and ELIXIR bio.tools metadata schemas">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">biocEDAM</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/biocEDAM.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/biotools.html">Examining biotools tagging of Bioconductor packages</a></li>
    <li><a class="dropdown-item" href="../articles/curate.html">Curation of Bioconductor package metadata, targeting EDAM ontology and ELIXIR bio.tools metadata schemas</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/vjcitn/biocEDAM/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="curate_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="curate_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="curate_files/jsoneditor-0.18.7/jsoneditor.min.js"></script><script src="curate_files/jsonedit-binding-4.0.0/jsonedit.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Curation of Bioconductor package metadata, targeting EDAM ontology and ELIXIR bio.tools metadata schemas</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">November 07, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/biocEDAM/blob/HEAD/vignettes/curate.Rmd"><code>vignettes/curate.Rmd</code></a></small>
      <div class="d-none name"><code>curate.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette is derived almost entirely from collaborative code supplied by Anh Nguyet Vu of Sage Bionetworks. The purpose is to illustrate usage of OpenAPI transformation to provide systematic organization and tagging of content available for Bioconductor packages.</p>
</div>
<div class="section level2">
<h2 id="example-1-chromvar">Example 1: chromVAR<a class="anchor" aria-label="anchor" href="#example-1-chromvar"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/biocEDAM">biocEDAM</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/timelyportfolio/listviewer" class="external-link">listviewer</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="va">docv</span> <span class="op">=</span> <span class="fu"><a href="../reference/curate_bioc.html">curate_bioc</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Success after 0 attempts</span></span>
<span><span class="co">## Success after 0 attempts</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">docv</span><span class="op">$</span><span class="va">base_final</span><span class="op">)</span><span class="op">)</span><span class="op">==</span><span class="fl">18</span><span class="op">)</span><span class="op">)</span> <span class="va">docv</span> <span class="op">=</span> <span class="fu"><a href="../reference/curate_bioc.html">curate_bioc</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># retry</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">docv</span><span class="op">$</span><span class="va">edam_processed</span><span class="op">)</span><span class="op">)</span><span class="op">==</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="va">docv</span> <span class="op">=</span> <span class="fu"><a href="../reference/curate_bioc.html">curate_bioc</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">docv</span><span class="op">$</span><span class="va">edam_processed</span><span class="op">)</span><span class="op">)</span><span class="op">==</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"curate_bioc failing on retry"</span><span class="op">)</span></span></code></pre></div>
<p>The curation step uses an OpenAI API key. The curated view of the chromVAR landing page at bioconductor.org is:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/listviewer/man/jsonedit.html" class="external-link">jsonedit</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">docv</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="jsonedit html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":{"name":"chromVAR","description":"Determine variation in chromatin accessibility across sets of annotations or peaks. Designed primarily for single-cell or sparse chromatin accessibility data, e.g. from scATAC-seq or sparse bulk ATAC or DNAse-seq experiments.","homepage":"http://bioconductor.org/packages/chromVAR/","version":"1.28.0","otherID":[{"type":"doi","value":"10.18129/B9.bioc.chromVAR","version":"1.28.0"}],"toolType":"Library","operatingSystem":["Linux","Windows","Mac"],"language":["R","C++"],"license":"MIT","collectionID":[],"maturity":"Mature","cost":"Free of charge","accessibility":"Open access","link":[{"url":"https://support.bioconductor.org/tag/chromvar","type":"Discussion forum","note":"Support site for chromVAR"},{"url":"https://bioconductor.org/checkResults/release/bioc-LATEST/chromVAR/","type":"Technical monitoring","note":"Build report for chromVAR"}],"download":[{"url":"http://bioconductor.org/packages/stats/bioc/chromVAR/","type":"Downloads page","note":"Download statistics for chromVAR","version":"1.28.0"}],"documentation":[{"url":"http://bioconductor.org/packages/chromVAR/","type":"User manual","note":"User manual for chromVAR"},{"url":"http://bioconductor.org/packages/chromVAR/","type":"Installation instructions","note":"Installation instructions for chromVAR"}],"publication":[{"doi":"10.18129/B9.bioc.chromVAR","type":"Primary","note":"Primary publication for chromVAR","version":"1.28.0"}],"credit":[{"name":"Alicia Schep","email":"aschep@gmail.com","typeEntity":"Person","typeRole":["Developer","Maintainer"]},{"name":"Jason Buenrostro","typeEntity":"Person","typeRole":"Contributor"},{"name":"Caleb Lareau","typeEntity":"Person","typeRole":"Contributor"},{"name":"William Greenleaf","typeEntity":"Person","typeRole":"Contributor"},{"name":"Stanford University","typeEntity":"Institute","typeRole":"Provider"}]},"options":{"mode":"tree","modes":["text","tree","table"]}},"evals":[],"jsHooks":[]}</script><p>The README of the chromVAR github repo provides additional metadata.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/listviewer/man/jsonedit.html" class="external-link">jsonedit</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">docv</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="jsonedit html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"data":{"topic":[{"term":"Bioinformatics","uri":"http://edamontology.org/topic_0091"},{"term":"Functional genomics","uri":"http://edamontology.org/topic_0085"},{"term":"Epigenomics","uri":"http://edamontology.org/topic_3173"},{"term":"Gene regulation","uri":"http://edamontology.org/topic_0204"}],"function":[{"operation":[{"term":"Sequence motif discovery","uri":"http://edamontology.org/operation_0238"},{"term":"Sequence motif recognition","uri":"http://edamontology.org/operation_0239"},{"term":"Gene expression profiling","uri":"http://edamontology.org/operation_0314"},{"term":"Differential binding analysis","uri":"http://edamontology.org/operation_3677"}],"input":[{"data":{"term":"Sequence record","uri":"http://edamontology.org/data_0849"},"format":[{"term":"BED","uri":"http://edamontology.org/format_3003"},{"term":"BAM","uri":"http://edamontology.org/format_2572"}]}],"output":[{"data":{"term":"Gene expression profile","uri":"http://edamontology.org/data_0928"},"format":[{"term":"CSV","uri":"http://edamontology.org/format_3752"}]}]}]},"options":{"mode":"tree","modes":["text","tree","table"]}},"evals":[],"jsHooks":[]}</script><p>These can be “merged” for a richer metadata artifact.</p>
</div>
<div class="section level2">
<h2 id="example-2-tximport">Example 2: tximport<a class="anchor" aria-label="anchor" href="#example-2-tximport"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dotx</span> <span class="op">=</span> <span class="fu"><a href="../reference/curate_bioc.html">curate_bioc</a></span><span class="op">(</span></span>
<span> <span class="st">"tximport"</span>, </span>
<span> devurl <span class="op">=</span> <span class="st">"https://raw.githubusercontent.com/thelovelab/tximport/refs/heads/devel/vignettes/tximport.Rmd"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Success after 0 attempts</span></span>
<span><span class="co">## Success after 0 attempts</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">full</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">dotx</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">dotx</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/listviewer/man/jsonedit.html" class="external-link">jsonedit</a></span><span class="op">(</span><span class="va">full</span><span class="op">)</span></span></code></pre></div>
<div class="jsonedit html-widget html-fill-item" id="htmlwidget-36aa3d2a04d42bbc2145" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-36aa3d2a04d42bbc2145">{"x":{"data":{"name":"tximport","description":"Imports transcript-level abundance, estimated counts and transcript lengths, and summarizes into matrices for use with downstream gene-level analysis packages. Average transcript length, weighted by sample-specific transcript abundance estimates, is provided as a matrix which can be used as an offset for different expression of gene-level counts.","homepage":"http://bioconductor.org/packages/tximport/","version":"1.34.0","otherID":[{"type":"doi","value":"10.18129/B9.bioc.tximport"}],"toolType":"Library","operatingSystem":["Linux","Windows","Mac"],"language":"R","license":"LGPL-2.0","collectionID":[],"maturity":"Mature","cost":"Free of charge","accessibility":"Open access","link":[{"url":"https://github.com/thelovelab/tximport","type":"Repository"},{"url":"https://support.bioconductor.org/tag/tximport","type":"Discussion forum"}],"download":[{"url":"http://bioconductor.org/packages/stats/bioc/tximport/","type":"Downloads page"}],"documentation":[{"url":"http://bioconductor.org/packages/tximport/","type":"User manual"},{"url":"http://bioconductor.org/packages/tximport/","type":"Installation instructions"}],"publication":[{"doi":"10.18129/B9.bioc.tximport","type":"Primary"}],"credit":[{"name":"Michael Love","email":"michaelisaiahlove@gmail.com","typeEntity":"Person","typeRole":["Maintainer","Developer"]},{"name":"Charlotte Soneson","typeEntity":"Person","typeRole":"Developer"},{"name":"Mark Robinson","typeEntity":"Person","typeRole":"Developer"},{"name":"Rob Patro","typeEntity":"Person","typeRole":"Contributor"},{"name":"Andrew Parker Morgan","typeEntity":"Person","typeRole":"Contributor"},{"name":"Ryan C. Thompson","typeEntity":"Person","typeRole":"Contributor"},{"name":"Matt Shirley","typeEntity":"Person","typeRole":"Contributor"},{"name":"Avi Srivastava","typeEntity":"Person","typeRole":"Contributor"}],"topic":[{"term":"Bioinformatics","uri":"http://edamontology.org/topic_0091"},{"term":"Transcriptomics","uri":"http://edamontology.org/topic_3308"},{"term":"Gene expression","uri":"http://edamontology.org/topic_0203"}],"function":[{"operation":[{"term":"RNA-Seq quantification","uri":"http://edamontology.org/operation_3800"},{"term":"Differential gene expression profiling","uri":"http://edamontology.org/operation_3223"},{"term":"Gene expression profiling","uri":"http://edamontology.org/operation_0314"}],"input":[{"data":{"term":"Sequence alignment","uri":"http://edamontology.org/data_0863"},"format":[{"term":"FASTA","uri":"http://edamontology.org/format_1929"},{"term":"FASTQ","uri":"http://edamontology.org/format_1930"},{"term":"SAM","uri":"http://edamontology.org/format_2573"},{"term":"BAM","uri":"http://edamontology.org/format_2572"}]}],"output":[{"data":{"term":"Gene expression profile","uri":"http://edamontology.org/data_0928"},"format":[{"term":"CSV","uri":"http://edamontology.org/format_3752"},{"term":"TSV","uri":"http://edamontology.org/format_3475"},{"term":"JSON","uri":"http://edamontology.org/format_3464"}]}]}]},"options":{"mode":"tree","modes":["text","tree","table"]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="example-3-msnbase">Example 3: MSnbase<a class="anchor" aria-label="anchor" href="#example-3-msnbase"></a>
</h2>
<p>We added a simpler function <code>edamize</code> to simply use chat completion to EDAM schema with input from a URL. This allows a setting of the completion “temperature” parameter, which defaults to zero. See <a href="https://gptcache.readthedocs.io/en/latest/bootcamp/temperature/chat.html" class="external-link">this doc</a>. <code>edamize</code> does not appear to be very robust and often returns Null in examples; work is needed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nn</span> <span class="op">=</span> <span class="fu"><a href="../reference/edamize.html">edamize</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/lgatto/MSnbase/refs/heads/master/vignettes/v04-benchmarking.Rmd"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Success after 0 attempts</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">full</span> <span class="op">=</span> <span class="fu"><a href="https://rstudio.github.io/reticulate/reference/r-py-conversion.html" class="external-link">py_to_r</a></span><span class="op">(</span><span class="va">nn</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/listviewer/man/jsonedit.html" class="external-link">jsonedit</a></span><span class="op">(</span><span class="va">full</span><span class="op">)</span></span></code></pre></div>
<div class="jsonedit html-widget html-fill-item" id="htmlwidget-febe03efa1a2d8d52a86" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-febe03efa1a2d8d52a86">{"x":{"data":{"topic":[{"term":"Proteomics","uri":"http://edamontology.org/topic_0121"},{"term":"Data visualisation","uri":"http://edamontology.org/topic_0092"},{"term":"Bioinformatics","uri":"http://edamontology.org/topic_0091"}],"function":[{"operation":[{"term":"Mass spectrum visualisation","uri":"http://edamontology.org/operation_3694"},{"term":"Protein quantification","uri":"http://edamontology.org/operation_3630"},{"term":"Data retrieval","uri":"http://edamontology.org/operation_2422"},{"term":"Spectral analysis","uri":"http://edamontology.org/operation_3214"}],"input":[{"data":{"term":"Mass spectrometry data","uri":"http://edamontology.org/data_2536"},"format":[{"term":"mzML","uri":"http://edamontology.org/format_3244"}]}],"output":[{"data":{"term":"Peptide identification","uri":"http://edamontology.org/data_0945"},"format":[{"term":"JSON","uri":"http://edamontology.org/format_3464"}]}]}]},"options":{"mode":"tree","modes":["text","tree","table"]}},"evals":[],"jsHooks":[]}</script>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vincent Carey.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
